library/psa_crypto_cipher.c

```diff
@@ -42,45 +42,85 @@ static psa_status_t mbedtls_cipher_validate_values(
     psa_algorithm_t alg,
     psa_key_type_t key_type)
 {
-    switch (alg) {
-        case PSA_ALG_STREAM_CIPHER:
-        case PSA_ALG_AEAD_WITH_SHORTENED_TAG(PSA_ALG_CHACHA20_POLY1305, 0):
-            if (key_type != PSA_KEY_TYPE_CHACHA20) {
-                return PSA_ERROR_NOT_SUPPORTED;
+    (void) alg; /* May be unused, depending on what algorithms/keys are enabled. */
+
+    switch (key_type) {
+#if defined(PSA_WANT_KEY_TYPE_CHACHA20)
+        case PSA_KEY_TYPE_CHACHA20:
+            switch (alg) {
+#if defined(PSA_WANT_ALG_STREAM_CIPHER)
+                case PSA_ALG_STREAM_CIPHER:
+#endif
+#if defined(PSA_WANT_ALG_CHACHA20_POLY1305)
+                case PSA_ALG_AEAD_WITH_SHORTENED_TAG(PSA_ALG_CHACHA20_POLY1305, 0):
+#endif
+            return PSA_SUCCESS;
```

Comment:
The indentation here looks so strange...

Reply:
Unfortunately `code_style.py` insists.